src/control/documents/mod.rs

```diff
@@ -1,89 +1,68 @@
 use std::fmt::Debug;
 
-use axum::{
-    extract::Path,
-    headers::{authorization::Bearer, Authorization},
-    Extension, Json, TypedHeader,
-};
-use http::StatusCode;
+use axum::{extract::Path, Extension, Json};
+use http::{HeaderMap, StatusCode};
 use tracing::info;
 
 use crate::{
-    domain::documents::{Document, DocumentData, DocumentMetaData},
-    repository::crud::{CrudCheckError, CrudRepository},
+    domain::documents::{Document, DocumentData},
     repository::documents::{
-        DocumentGetError, DocumentPostError, DocumentPutError, DocumentRepository,
+        DocumentGetError, DocumentInsertError, DocumentRepository, DocumentUpdateError,
     },
 };
 
 #[tracing::instrument(skip_all)]
-pub async fn get_documents<T: DocumentRepository + CrudRepository + Clone + Send + Sync>(
+pub async fn get_documents<T: DocumentRepository + Clone + Send + Sync>(
     Extension(repository): Extension<T>,
     Path(project_id): Path<i32>,
-    TypedHeader(user_id): TypedHeader<Authorization<Bearer>>,
+    _headers: HeaderMap,
```

Comment:
As I wrote in another comment, I think the correct way to extract `X-User-Id` will be to create a struct `XUserId(i32)` that implements `Header` then you can use `TypedHeader(user_id)` here easily,

Reply:
Header created